core/src/main/kotlin/com/alecstrong/sql/psi/core/SqlCoreEnvironment.kt

```diff
@@ -59,9 +61,15 @@ private object ApplicationEnvironment {
   }
 }
 
+data class PredefinedTable(val packageName: String, val simpleFileName: String, val content: String) {
+  val fileName = "$packageName.$simpleFileName"
+}
+
 open class SqlCoreEnvironment(
   sourceFolders: List<File>,
   dependencies: List<File>,
+  predefinedTables: List<PredefinedTable>,
```

Comment:
I think a lot of this becomes cleaner if this is a `List<File>` and you just do the file reads inside of the constructor

Reply:
got it, that makes sense. Does this even need to be passed into sql-psi? Could we just say it has to follow some specific format `system_tables.predefined` or whatever and then we can just check if its in the resources directory?

Don't feel ridiculously strongly on this either way

Reply:
How do you resolve the references without sql-psi? And psi needs files..., wait, how does IntelliJ index dependency jars? 🤔

Reply:
But this function still needs strings, so the signature is correct. How do you get the strings, eg via the resources should be implemented in the dialect, because it is in their jar file.

Reply:
it should all be loaded on to the same class loader, so you can reference resources from another jar

classloaders are insane so I am definitely not 100% confident in that

Reply:
Hm, I am about to adopt your comments but honestly, I don't see, what you/we want to change. We can't use `File`. While this is the most idiomatic type, this won't work with resource files. So we need to pass the content of the file. In the beginning, I just used to use `List<String>` and named the VirtualLightFile by its index: `1.predefined` with a hardcoded suffix. But this makes debugging way harder because the order of the files loaded at runtime by the jar is not deterministic. 
So the current idea/implementation is to load the predefined files by the dialect, in the best case using the file name from the resources. sql-psi does not care about loading nor naming etc.

Reply:
If we want some kind of autoloading some ressource files we should either provide an extension or simple move this function to sqldelight dialect api, for example. I would not add it to the sql environment because you could use `sql-psi` without sqldelight, like I do in my frontend.